File: iOS/src/Payment.m
//
//  Payment.m
//  GL Wrapper IOS
//
//  Created by SuperDan on 12/1/13.
//  Copyright (c) 2013 SuperDan. All rights reserved.
//

#import "Payment.h"

@implementation Payment
- (id) init: (NSString*) newVersion
           : (NSString *) newToCharge
           : (NSString *) newToken
{
    version = newVersion;
    toCharge = newToCharge;
    token = newToken;
    description = @"";
    return self;
}
- (id) setDescription: (NSString *) desc
{
    description = desc;
    
    return self;
}
- (NSString *) charge: (float) amount
                     : (NSString *) refNo
{
    NSMutableDictionary *dictionary = [[NSMutableDictionary alloc] init];
    [dictionary setObject: token forKey:@"access_token"];
    [dictionary setObject: toCharge forKey:@"endUserId"];
    [dictionary setObject: [NSString stringWithFormat:@"%.02f", amount] forKey:@"amount"];
    [dictionary setObject: refNo forKey:@"referenceCode"];
    [dictionary setObject: @"charged" forKey:@"transactionOperationStatus"];
    [dictionary setObject: description forKey:@"description"];
    
    return [self curlPost:@"http://devapi.globelabs.com.ph/payment/v1/transactions/amount"
                         :dictionary];
}
@end

File: Android/src/com/globelabs/api/Payment.java
package com.globelabs.api;

import java.util.ArrayList;
import java.util.List;

import org.apache.http.NameValuePair;
import org.apache.http.message.BasicNameValuePair;

public class Payment extends GlobeApi{
	protected String version;
	protected String toCharge;
	protected String token;
	protected String description;
	protected String clientCorrelator;
	
	public Payment(String version, String toCharge, String token) {
		this.version = version;
		this.toCharge = toCharge;
		this.token = token;
	}

	public void setVersion(String version) {
		this.version = version;
	}
	
	public String getToCharge() {
		return toCharge;
	}
	
	public void setToCharge(String toCharge) {
		this.toCharge = toCharge;
	}

	public String getToken() {
		return token;
	}

	public void setToken(String token) {
		this.token = token;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getClientCorrelator() {
		return clientCorrelator;
	}

	public void setClientCorrelator(String clientCorrelator) {
		this.clientCorrelator = clientCorrelator;
	}

	public void charge(float amount, String refNo, PostRequestHandler handler) {
		List<NameValuePair> parameters = new ArrayList<NameValuePair>();
		parameters.add(new BasicNameValuePair("endUserId", this.toCharge));
		parameters.add(new BasicNameValuePair("amount", String.format("%.2f", amount)));
		parameters.add(new BasicNameValuePair("description", this.description));
		parameters.add(new BasicNameValuePair("referenceCode", refNo));
		parameters.add(new BasicNameValuePair("transactionOperationStatus", "charged"));
		parameters.add(new BasicNameValuePair("access_token", this.token));
		String url = "https://%s/payment/%s/transactions/amount";
		url = String.format(url, GlobeApi.API_ENDPOINT, "v1");
		this.__curlPost(url, parameters, handler);
	}
}
